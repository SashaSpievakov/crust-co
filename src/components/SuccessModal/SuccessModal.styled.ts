import styled from 'styled-components';

import { ButtonBack } from 'src/styles/Buttons.styled';

const Aricle = styled.article`
  position: fixed;
  top: 40%;
  left: 50%;
  width: 40%;
  height: 30%;
  text-align: center;
  background: ${({ theme }) => theme.colors.fourth};
  transform: translate(-50%, -40%);
  z-index: 10;
  border-radius: 15px;
  box-shadow: 0 5px 20px 0 rgba(0, 0, 0, 0.05);

  @media screen and (max-width: 900px) {
    width: 70%;
    height: 40%;
  }

  @media screen and (max-width: 500px) {
    top: 35%;
    width: 80%;
    transform: translate(-50%, -35%);
  }
`;

const Header = styled.h3`
  font-size: 2.5rem;
  margin-top: 5%;

  @media screen and (max-width: 700px) {
    font-size: 2rem;
  }

  @media screen and (max-width: 500px) {
    font-size: 1.7rem;
  }
`;

const SubHeader = styled.h4`
  font-size: 1.3rem;
  font-weight: 400;
  width: 60%;
  margin: 2% auto 0;

  @media screen and (max-width: 700px) {
    font-size: 1rem;
  }
`;

const CloseButton = styled(ButtonBack)`
  margin-top: 5%;
`;

export { Aricle, Header, SubHeader, CloseButton };
