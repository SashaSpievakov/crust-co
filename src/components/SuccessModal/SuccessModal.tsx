import { FC } from 'react';
import { ModalBg } from 'src/styles/Base.styled';

import { Aricle, CloseButton, Header, SubHeader } from './SuccessModal.styled';

interface SuccessModalProps {
  setIsOpen: (value: boolean) => void;
}

export const SuccessModal: FC<SuccessModalProps> = ({ setIsOpen }) => {
  return (
    <>
      <ModalBg onClick={() => setIsOpen(false)} />
      <Aricle>
        <Header>Thank you for your order!</Header>
        <SubHeader>
          Our operator will give you a call in a few minutes
        </SubHeader>
        <CloseButton onClick={() => setIsOpen(false)}>Close</CloseButton>
      </Aricle>
    </>
  );
};
