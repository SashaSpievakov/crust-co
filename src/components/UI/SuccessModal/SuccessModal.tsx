import { ModalBg } from 'src/styles/Base.styled';
import { Aricle, Header, SubHeader, CloseButton } from './SuccessModal.styled';

interface SuccessModalProps {
  setIsOpen: (value: boolean) => void;
}

const SuccessModal = ({ setIsOpen }: SuccessModalProps) => {
  return (
    <>
      <ModalBg onClick={() => setIsOpen(false)} />
      <Aricle>
        <Header>Thank you for your order!</Header>
        <SubHeader>
          Our operator will give you a call in a few minutes
        </SubHeader>
        <CloseButton onClick={() => setIsOpen(false)}>
          <span>Close</span>
        </CloseButton>
      </Aricle>
    </>
  );
};
export default SuccessModal;
