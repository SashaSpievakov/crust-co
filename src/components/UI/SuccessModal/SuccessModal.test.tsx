import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { CartItemsMockProps } from 'src/tests/mocks/mockData/mockData';
import SuccessModal from './SuccessModal';
import rendererWithAllProviders from '../../../tests/helpers/rendererWithProviders';
import renderWithProviders from '../../../tests/helpers/renderWithProviders';

describe('SuccessModal Tests', () => {
  test('renders the SuccessModal UI component', () => {
    const snapshot = rendererWithAllProviders(
      <SuccessModal setIsOpen={jest.fn()} />,
    );
    expect(snapshot).toMatchSnapshot();
  });

  test('closes the SuccessModal', async () => {
    renderWithProviders(null, true, '/cart', {
      cart: CartItemsMockProps,
    });
    const buyButton = screen.getByRole('button', {
      name: /buy now/i,
    });

    userEvent.click(buyButton);

    const sumbitBtn = screen.getByRole('button', {
      name: /send/i,
    });
    const inputs = screen.getAllByRole('textbox');
    const phoneInput = screen.getByRole('spinbutton');

    userEvent.type(phoneInput, '4730395859');
    userEvent.type(inputs[0], 'Alex');
    userEvent.type(inputs[1], 'Toronto');
    userEvent.type(inputs[2], 'some address');
    userEvent.click(sumbitBtn);

    const returnBtn = await screen.findByText(/close/i);

    userEvent.click(returnBtn);

    expect(
      screen.queryByText(/thank you for your order!/i),
    ).not.toBeInTheDocument();
  });
});
