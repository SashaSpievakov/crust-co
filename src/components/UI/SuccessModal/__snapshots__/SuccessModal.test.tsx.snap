// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal Tests renders the SuccessModal UI component 1`] = `
Object {
  "0": <div
    className="sc-dkrFOg ewVMXk"
    onClick={[Function]}
  />,
  "1": <article
    className="sc-kDvujY cnQXHJ"
  >
    <h3
      className="sc-ipEyDJ gtVqqb"
    >
      Thank you for your order!
    </h3>
    <h4
      className="sc-csuSiG kPjyyr"
    >
      Our operator will give you a call in a few minutes
    </h4>
    <button
      className="sc-hLBbgP sc-eDvSVe sc-eDWCr hEWbyi eUGSku tncLY"
      onClick={[Function]}
    >
      <span>
        Close
      </span>
    </button>
  </article>,
}
`;
